src/Service/ConfigurationService.php

```diff
@@ -124,22 +126,18 @@ public function getHmacLive()
      */
     public function getHmacKey()
     {
-        if ($this->getEnvironment()) {
-            return $this->getHmacLive();
-        }
-
-        return $this->getHmacTest();
+        return $this->systemConfigService->get(
+            self::BUNDLE_NAME . '.config.hmac' . ucfirst($this->getEnvironment())
+        );
```

Comment:
I think the previous approach was much easier to understand. Now it's a bit complicated for just a simple config retriever.
I think we can still do something like

```suggestion
        if ($this->getEnvironment() === Environment::LIVE) {
            return $this->getHmacLive();
        )
        
        return $this->getHmacTest();
```